### PR TITLE
fluent-bit: init at 1.0.6

### PR DESCRIPTION
**Pull request by samrose**
###### Motivation for this change
adding a package that installs https://fluentbit.io

###### Things done
created a pkg, and added to all-packages.nix

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ pkgs, stdenv, fetchgit, cmake, ... }:
+
+stdenv.mkDerivation rec {
+  src = fetchgit {
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: Please use fetchFromGitHub instead of fetchgit

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ pkgs, stdenv, fetchgit, cmake, ... }:
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: Please do not include `pkgs` as an argument in a derivation. There is also no need for `...` in this case.

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+stdenv.mkDerivation rec {
+  src = fetchgit {
+    url = https://github.com/fluent/fluent-bit;
+    sha256 = "0rmdbrhhrim80d0hwbz56d5f8rypm6h62ks3xnr0b4w987w10653";
+    rev = "8cc3a1887c3fcd6dd95a4a475fb213a0e399c222";
+  };
+  name = "fluentbit";
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: Please change from `name` to `pname`.

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  version = "1.0.6";
+  buildInputs = [ cmake ];
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: Please use `nativeBuildInputs` instead of `buildInputs` for `cmake`.

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  builder = ./builder.sh;
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: I'm not seeing a specific reason you need your own builder so you can remove this and the default will take care of it for you.

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  meta = with stdenv.lib; {
+    description = "fluentbit";
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: The description should be not include the name of the software but instead describe what the software is/does.

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    homepage = "https://fluentbit.io";
+    maintainers = [ maintainers.samrose ];
+    license = licenses.asl20;
+    platforms = platforms.linux;
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: The website says this software also supports osx.

File: pkgs/tools/misc/fluent-bit/default.nix [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  };
+  name = "fluentbit";
+  version = "1.0.6";
+  buildInputs = [ cmake ];
+  builder = ./builder.sh;
+
+  meta = with stdenv.lib; {
+    description = "fluentbit";
+    homepage = "https://fluentbit.io";
+    maintainers = [ maintainers.samrose ];
```
Comment (aanderse) [inline pkgs/tools/misc/fluent-bit/default.nix]: You'll need to add yourself as an entry to maintainers/maintainer-list.nix, though please do this in a separate commit.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/fluent-bit/default.nix (hunk 6 above)
Comment (JohnAZoidberg) [reply pkgs/tools/misc/fluent-bit/default.nix]: E.g.: "Multi-platform Log Processor and Forwarder which allows you to collect data/logs from different sources, unify and send them to multiple destinations"
Maybe just that: "Multi-platform Log Processor and Forwarder" and the rest as `longDescription`

Comment (jtojnar) [reply pkgs/tools/misc/fluent-bit/default.nix]: Even the information that it supports multiple platforms is redundant. We have `platforms` field for that.

Comment (aanderse) [reply pkgs/tools/misc/fluent-bit/default.nix]: Good point.

File: pkgs/top-level/all-packages.nix [hunk 9]
```diff
@@ -2824,6 +2824,8 @@ in
 
   fltrdr = callPackage ../tools/misc/fltrdr { stdenv = gcc8Stdenv; };
 
+  fluentbit = callPackage ../tools/misc/fluent-bit { };
```
Comment (yegortimoshenko) [inline pkgs/top-level/all-packages.nix]: Package name should be consisent with folder name, derivation name, and preferably, executable name. `fluent-bit` seems to be the name chosen by upstream.

File: pkgs/tools/misc/fluent-bit/default.nix (hunk 5 above)
Comment (yegortimoshenko) [reply pkgs/tools/misc/fluent-bit/default.nix]: The package does not trivially build with the default builder. However, there are two problems with this builder.

1. `$out/bin` is filled with programs that do not belong in the package (`xxd-c`, `hello_world`, `onigmo-config`, etc.) while `$out/{include,lib}` is missing.

2. Builders do not compose and generally should be used as a last resort.

In this case, upstream assumed FHS location of systemd unit directory, which is read-only on NixOS, which causes an error during `installPhase`. Fixing that in `postPatch` phase will suffice:

```
  postPatch = ''
    substituteInPlace src/CMakeLists.txt \
      --replace /lib/systemd $out/lib/systemd
  '';
```

**Review by lukateras — CHANGES_REQUESTED** [review]
Please review changes between b9f68c5f0fe3088c73cc4587063d7b683c3f2f2d and 71077639c833763edb3e9b255706794fbc7e9b4a.

Also, you will need to add yourself to maintainers list in order for metadata to evaluate correctly (as a separate commit). See instructions in [`maintainers/maintainer-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L1-L38) on how to do so.

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build fluent-bit

**Review by lukateras — APPROVED** [review]

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build fluent-bit

**Comment by aanderse** [comment]
It would be nice if the git history was rewritten here into 2 commits, the first of which adding samrose to the maintainer list, and the second being the rest squashed.
